### PR TITLE
docs(best-practices): prefer `satisfies` operator over `as`

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Prefer `satisfies` operator over `as`
- Link to the TypeScript release-notes for the version that has the `satisfies` operator feature

## Context

Helps with issue:

- #19856

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/development/best-practices.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ Read the [GitHub Docs, renaming a branch](https://docs.github.com/en/repositorie
   - Only use `istanbul` comments for unreachable code coverage that is needed for `codecov` completion
   - Use descriptive `istanbul` comments
 - Avoid cast or prefer `x as T` instead of `<T>x` cast
+- Prefer `satisfies` operator over `as`, read the [TypeScript release notes for `satisfies` operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html#the-satisfies-operator) to learn more
```
Comment (HonkingGoose) [inline docs/development/best-practices.md:35]: The `## General` list seems like a good place for this. Let me know if we need:

- to put the `satisfies` recommendation in it's own section?
- to have a Renovate specific example?
- to explain better why we prefer `satisfies` over `as`?

**Comment by rarkins** [comment]
Ideally all these types of best practices would be enforced by linting rules, and even better yet auto fixable

File: docs/development/best-practices.md [hunk 2]
```diff
@@ -32,6 +32,7 @@ Read the [GitHub Docs, renaming a branch](https://docs.github.com/en/repositorie
   - Only use `istanbul` comments for unreachable code coverage that is needed for `codecov` completion
   - Use descriptive `istanbul` comments
 - Avoid cast or prefer `x as T` instead of `<T>x` cast
```
Comment (HonkingGoose) [inline docs/development/best-practices.md:34]: @viceice can you make a suggestion  to update this line?

Comment (viceice) [reply docs/development/best-practices.md:34]: not sure, maybe leave as is for now

**Review by viceice — APPROVED** [review]

**Comment by HonkingGoose** [comment]
> Ideally all these types of best practices would be enforced by linting rules, and even better yet auto fixable

I agree! Let's talk about it in this issue:

- #20612

**Pull request merged**